pkg/cache/task_state.go

```diff
@@ -314,6 +314,7 @@ func TaskStates() *TStates {
 	return storeTaskStates
 }
 
+//nolint:funlen
 func newTaskState() *fsm.FSM {
```

Comment:
this is production code, so maybe we can do a bit refactor? for example, we can add two helper methods to generate `eventDesc` and `callbacks`. With those helpers, `newTaskState` can be simplified to following style
```go
func newTaskState() *fsm.FSM {
	states := TaskStates()
	return fsm.NewFSM(states.New, eventDesc(states), callbacks(states))
}

```

Reply:
Ok! no problem.